### PR TITLE
T250218: Prompt users for consent to enable app instrumentation

### PR DESCRIPTION
**Pull request by medied**
🚧 DRAFT PR

Phabricator Link: https://phabricator.wikimedia.org/T250218

### Note

Sister task info:

https://phabricator.wikimedia.org/T250219
https://github.com/wikimedia/wikipedia-kaios/pull/218

---

TODOs:

- [ ] fix tests cases
- [ ] more testing on device

### CODE REVIEW

File: src/components/Search.js [hunk 1]
```diff
@@ -19,12 +19,39 @@ const SearchOfflinePanel = () => {
   )
 }
 
+const ConsentPopup = ({ close, i18n }) => {
```
Comment (stephanebisson) [inline src/components/Search.js:22]: Please move to its own file.

File: src/components/Search.js [hunk 2]
```diff
@@ -19,12 +19,39 @@ const SearchOfflinePanel = () => {
...
+  }
+
+  useSoftkey('ConsentMessage', {
+    right: i18n('softkey-yes'),
+    onKeyRight,
+    left: i18n('softkey-no'),
+    onKeyLeft
+  }, [])
+
+  return <div class='consent-prompt-message'>
```
Comment (stephanebisson) [inline src/components/Search.js:42]: style
```
return (
  <Components ... />
)
```

File: src/components/Search.js [hunk 3]
```diff
@@ -57,6 +84,9 @@ export const Search = () => {
   useEffect(() => {
     articleHistory.clear()
     setNavigation(0)
+    if (getConsentConfirmationStatus() === false) {
+      showConsentPopup({ i18n })
```
Comment (stephanebisson) [inline src/components/Search.js:88]: `ConsentPopup` should `useI18n()` on its own.

File: src/components/Search.js [hunk 4]
```diff
@@ -19,12 +19,39 @@ const SearchOfflinePanel = () => {
   )
 }
 
+const ConsentPopup = ({ close, i18n }) => {
+  const onKeyRight = () => {
```
Comment (stephanebisson) [inline src/components/Search.js:23]: `onKeyRight` and `onKeyLeft` are basically identical except for one value. You can explore using a factory function to build handlers preconfigured for a specific response. Something like:
```
const handleConsent = response => {
  return () => {
    setConsentStatus(response)
    setConsentConfirmationStatus(true)
    close()
  }
}
...
onKeyRight: handleConsent(true),
onKeyLeft: handleConsent(false),
```

File: src/utils/usageReportConsent.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const KEY = 'usage-data-consent'
+const confirmKey = 'usage-data-consent-confirmed-once'
```
Comment (stephanebisson) [inline src/utils/usageReportConsent.js:2]: If you want to keep this, consider using the same casing for both keys.

You could also consider using a single key (`usage-data-consent`) with the following values:
`true`: the user accepts
`false`: the user refuses
`null or undefined`: the user has not specified any preference yet
I would still keep `getConsentConfirmationStatus()` to check for null/undefined for proper encapsulation.

**Comment by medied** [comment]
Closing this PR in favor of https://github.com/wikimedia/wikipedia-kaios/pull/220

**Pull request closed without merging**